source/v5/mqtt3_to_mqtt5_adapter.c

```diff
@@ -1668,3 +1729,118 @@ struct aws_mqtt_client_connection *aws_mqtt_client_connection_new_from_mqtt5_cli
 
     return &adapter->base;
 }
+
+#define DEFAULT_MQTT_ADAPTER_OPERATION_TABLE_SIZE 100
+
+void aws_mqtt3_to_mqtt5_adapter_operation_table_init(
+    struct aws_mqtt3_to_mqtt5_adapter_operation_table *table,
+    struct aws_allocator *allocator) {
+    aws_mutex_init(&table->lock);
+    aws_hash_table_init(
+        &table->operations,
+        allocator,
+        DEFAULT_MQTT_ADAPTER_OPERATION_TABLE_SIZE,
+        aws_mqtt_hash_uint16_t,
+        aws_mqtt_compare_uint16_t_eq,
+        NULL,
+        NULL);
+    table->next_id = 1;
+}
+
+static int s_adapter_operation_clean_up(void *context, struct aws_hash_element *operation_element) {
+    (void)context;
+
+    struct aws_mqtt3_to_mqtt5_adapter_operation_base *operation = operation_element->value;
+
+    aws_mqtt3_to_mqtt5_adapter_operation_destroy(operation);
+
+    return AWS_COMMON_HASH_TABLE_ITER_CONTINUE;
+}
+
+void aws_mqtt3_to_mqtt5_adapter_operation_table_clean_up(struct aws_mqtt3_to_mqtt5_adapter_operation_table *table) {
+    aws_hash_table_foreach(&table->operations, s_adapter_operation_clean_up, table);
+
+    aws_hash_table_clean_up(&table->operations);
```

Comment:
should we run `aws_mutex_clean_up` here? The init and clean up of a mutex is different across platforms and I'm seeing places we run clean up and places we don't so maybe it's unnecessary?

Reply:
Yes